pkg/config/config.go

```diff
@@ -33,22 +40,67 @@ func GetConfigs(filename string) (*Configs, error) {
 		log.Fatal(err)
 	}
 
+	if err := parseEnv(&configs); err != nil {
+		log.Fatal(err)
+	}
+
 	return &configs, nil
 }
 
-// GetAllImages extracts a set of images required for all the tasks to run
-func (c *Configs) GetAllImages() ([]string, error) {
-	var images []string
-	var set = make(map[string]bool)
+func parseEnv(configs *Configs) error {
+	file := viper.GetString("DotenvFile")
+	envs, err := godotenv.Read(file)
+	if err != nil {
+		log.Warn(err)
+	}
 
-	for _, tasks := range c.Tasks {
-		for _, task := range tasks {
-			set[task.Image] = true
+	for k, tasks := range (*configs).Tasks {
+		for j, task := range tasks {
+			for i, envVar := range task.Envs {
```

Comment:
Shall we make this part into a separate function, may be like `replaceEnvVarsWithValues`?

Reply:
I know the function is quite big and doesn't looks good. But there is no point in calling out a different function from line 57 of [`config.go`](https://github.com/leopardslab/Dunner/pull/43/files#diff-091ace02b49a46f55f2586cfec89a160) because then the sole purpose of `parseEnv` function would be to read dotenv file and call another function. Also, a separate function cannot be called from line 60 (after three-tier `for loops`) because there I have to pass many arguments, viz., `envs`, `i`,`j`,`k`, `envVar`, `configs`, which would again defeat the purpose.